client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -92,18 +169,76 @@ class SpeedUpSiteSettings extends Component {
 	}
 }
 
-export default connect( state => {
-	const selectedSiteId = getSelectedSiteId( state );
-	const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
-	const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
-		state,
-		selectedSiteId,
-		'photon'
-	);
+export default connect(
+	state => {
+		const selectedSiteId = getSelectedSiteId( state );
+		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
+		const siteAcceleratorSupported = isJetpackMinimumVersion( state, selectedSiteId, '6.6-alpha' );
+		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
+			state,
+			selectedSiteId,
+			'photon'
+		);
+		const photonModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon' );
+		const assetCdnModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon-cdn' );
+		const isPhotonActivating = isActivatingJetpackModule( state, selectedSiteId, 'photon' );
+		const isAssetCdnActivating = isActivatingJetpackModule( state, selectedSiteId, 'photon-cdn' );
+		const isPhotonDeactivating = isDeactivatingJetpackModule( state, selectedSiteId, 'photon' );
+		const isAssetCdnDeactivating = isDeactivatingJetpackModule(
+			state,
+			selectedSiteId,
+			'photon-cdn'
+		);
+
+		let togglingSiteAccelerator;
+		// First Photon activating.
+		if ( isPhotonActivating ) {
+			if ( assetCdnModuleActive ) {
+				togglingSiteAccelerator = false;
+			} else {
+				togglingSiteAccelerator = true;
+			}
+			// Then Asset CDN activating.
+		} else if ( isAssetCdnActivating ) {
+			if ( photonModuleActive ) {
+				togglingSiteAccelerator = false;
+			} else {
+				togglingSiteAccelerator = true;
+			}
+			// Then Photon deactivating.
+		} else if ( isPhotonDeactivating ) {
+			if ( assetCdnModuleActive ) {
+				togglingSiteAccelerator = false;
+			} else {
+				togglingSiteAccelerator = true;
+			}
+			// Then Asset CDN deactivating.
+		} else if ( isAssetCdnDeactivating ) {
+			if ( photonModuleActive ) {
+				togglingSiteAccelerator = false;
+			} else {
+				togglingSiteAccelerator = true;
+			}
+		} else {
+			togglingSiteAccelerator = false;
+		}
```

Comment:
Could we get rid of this specificity now that we're using `updateFields()`? I think it would make sense to use the `isSavingSettings` and `isRequestingSettings` props that are provided by `wrapSettingsForm`. This way we wouldn't need our own `togglingSiteAccelerator` prop, and all the UI controls would get disabled consistently when we touch any of the toggles. 

What do you think?

Reply:
I am afraid we still need this. The problem isn't really about the disabled status of the toggle; as you said this can be handled anytime the settings are being saved. This helps handling the toggling status of the main toggle, so it turns from on to off and off to on.

If we simplify this to change the toggling status every time a setting is changed, there will be cases where the toggle status switches to off although the toggle should stay on.

Here is an example:

- one module is already on, so the main toggle is on.
- You are turning the second module on. The module should stay on, so the toggling status should not be triggered again.

Reply:
I understand what you mean, but I'm not sure we're talking about the same thing.

The thing here is that we don't really use the `toggling` prop much in site settings. And I'm only suggesting that we make everything consistent with the rest of the site settings in Calypso. I don't think we need to specify the `toggling` at all. Whether a field is on or off, or whether it's currently being fetched/updated are really different things.

If you have a look at any other compact form toggles in site settings, you'll notice `toggling` isn't used at all. Instead, we often do `disabled={ isRequestingSettings || isSavingSettings }`, which disables the toggle while any requesting or saving is in progress. Because of previous logic and versions of the code, we never had the chance to achieve good behavioral consistency throughout all the site settings UI, but at least we can try to do it for any future UI.

In addition to visual and functional consistency, I'm suggesting this to reduce the chance for bugs. When disabling all the UI fields when updating a setting, we prevent from overwriting previously set settings, and we protect from race conditions and scenarios like this:

* User enables toggle of setting X.
* Setting X request is dispatched.
* User enables toggle of setting Y.
* Setting Y request is dispatched.
* Setting Y request is completed.
* Setting X request is completed.

Since the site settings request contains all the modules and all their settings, the data from the request for X might override the data for the request on Y on the server, while the client would think otherwise. 

FWIW, activating or deactivating a module currently uses a different endpoint, so when activating/deactivating a module, we won't disable the UI if we don't specifically check for that.

So, in practice, while I don't advocate completely removing our reliance on (de)activation for practical reasons, I suggest the following:

* Not using the `toggling` prop at all.
* Replacing the long and repetitive logic with something as simple as `const togglingSiteAccelerator = isPhotonActivating || isPhotonDeactivating || isAssetCdnActivating || isAssetCdnDeactivating;`

Would you like to give that a shot and let me know what you think?

Reply:
I've removed the toggling prop and it doesn't really impair the UX in my opinion, so let's go with that 👍 	
1dabfe91664423f8ab76cfedfac3fef99d606921